### PR TITLE
Clean up task output message handling

### LINKED ISSUE #1604: Task prerequisite and output messages don't need to contain the task ID
There's no longer any need for task prerequisites and output messages to contain the task ID (originally there was I think, pre dependency graph). The graph shows which tasks are depended on, and the suite daemon knows the sender of incoming task messages.

### PR DESCRIPTION
**Pull request by hjoliver**
Close #1604.

 * Task ID is no longer prepended to output messages at the client end and sent across the network (not necessary as the receiving object in the server has the same ID).
 * Task ID is now only prepended to output messages inside the data structure handed to the broker to match with prerequisites.
 * Treat message outputs the same as standard outputs - TaskID prepended as per the previous bullet point.
 * Deprecated "message offset" placeholders in message trigger strings (see below https://github.com/cylc/cylc/pull/1761#issuecomment-196574461)
 * As a consequence of the previous points, simplify output message handling in `task_proxy.py` by getting rid of a bunch of Task ID prepend/remove operations.
 * Eliminates some annoying redundancy in the suite log: `[task-id] ... task-id succeeded` is now just `[task-id] ...succeeded`, etc.
 * Rename some associated modules and classes (e.g. `output.py:output` to `message_output.py:MessageOutput` - as this class is only used for message outputs).

### CODE REVIEW

**Comment by hjoliver** [comment]
(test battery is passing on my laptop VM)

**Comment by hjoliver** [comment]
Re deprecation of cycle point offset placeholders in message trigger strings ... this:
```
[[[P1M]]]
   graph = foo[-P1M]:out1 => bar
...
[runtime]
   [[foo]]
       script = """
cylc message "files ready for bar at $(cylc cyclepoint --offset=P1M)"
"""
       [[[outputs]]]
             out1 = "output files ready for bar at [P1M]"
```
is entirely equivalent to this:
```
[[[P1M]]]
   graph = foo[-P1M]:out1 => bar
...
[runtime]
   [[foo]]
       script = cylc message "files ready for bar"
       [[[outputs]]]
             out1 = "files ready for bar"
```

On master, we enforce the use of the offset placeholders in message output strings, in order to  distinguish the message strings emitted by successive instances of `foo`.  But this was only necessary because we were not prepending task ID to message strings as is automatically done for the standard outputs (succeeded etc.).  Wracking my brains to recall how this crazy situation came about, I believe it goes right back to cylc-2 when there was no dependency graph (and therefore no cycle point offset notation in the graph), so this was the way to implement inter-cycle triggering in the case of message triggers. 

I've change the documentation to get rid of all references to these in-string offset placeholders, but it's still supported in the code for backward compatibility.

File: lib/cylc/trigger.py [hunk 1]
```diff
@@ -24,8 +25,43 @@
...
+            if not warned:
+                print >> sys.stderr, DEPRECATION_TEMPLATE % msg
+                warned = True
+            prefix, signed_offset, sign, offset, suffix = n.groups()
+            if offset:
+                offset = get_interval(signed_offset)
+            else:
+                offset = get_interval_cls().get_null()
+        # else: Plain message, no offset.
+    return offset
```
Comment (hjoliver) [inline lib/cylc/trigger.py]: This code was duplicated in two places.

File: lib/cylc/trigger.py [hunk 2]
```diff
@@ -78,53 +114,34 @@ def __init__(
...
-                        msg_offset = get_interval_cls().get_null()
-                else:
-                    raise TriggerError(
-                        "ERROR: undefined trigger qualifier: %s:%s" % (
-                            task_name, qualifier))
-            self.message = msg
-            self.message_offset = msg_offset
-
-    def is_standard(self):
-        return self.builtin is not None
```
Comment (hjoliver) [inline lib/cylc/trigger.py]: method not used anywhere.

File: lib/cylc/trigger.py [hunk 3]
```diff
@@ -133,4 +150,4 @@ def get_prereq(self, point):
                 point = get_point_relative(
                     self.graph_offset_string, point)
             preq = TaskID.get(self.task_name, point) + ' ' + self.builtin
-        return preq, point
+        return preq
```
Comment (hjoliver) [inline lib/cylc/trigger.py]: The second return value was not used anywhere.

File: tests/message-triggers/03-placeholder/suite.rc [hunk 4]
```diff
@@ -1,12 +0,0 @@
...
-    initial cycle point = 1
-    [[dependencies]]
-        [[[P1]]]
-            graph = foo:x => bar
-[runtime]
-    [[foo]]
-        script = cylc message "hello to $CYLC_TASK_CYCLE_POINT"
-        [[[outputs]]]
-            x = "hello to Bob"
-            # (should be "hello to []")
```
Comment (hjoliver) [inline tests/message-triggers/03-placeholder/suite.rc]: (not anymore, it shouldn't!)

**Comment by hjoliver** [comment]
@matthewrmshin - please review or re-assign.  Test battery passes here.

**Comment by matthewrmshin** [comment]
`./tests/message-triggers/03-new-style.t` is failing due to timezone issue.

File: lib/cylc/task_trigger.py [hunk 5]
```diff
@@ -24,8 +25,50 @@
...
+    TODO - this function can be deleted once the deprecated cycle point offset
+    placeholders are removed from cylc (see GitHub #1761).
+
+    """
+
+    offset = None
+    global warned
+    global BCOMPAT_MSG_RE_C5
+    global BCOMPAT_MSG_RE_C6
+    global DEPRECN_WARN_TMPL
```
Comment (matthewrmshin) [inline lib/cylc/task_trigger.py]: I don't think you need to declare `global` for constants.

File: lib/cylc/task_trigger.py [hunk 6]
```diff
@@ -24,8 +25,50 @@
...
+
+    # cylc-5 [T+n] message offset - DEPRECATED
+    m = re.match(BCOMPAT_MSG_RE_C5, msg)
```
Comment (matthewrmshin) [inline lib/cylc/task_trigger.py]: You can do `BCOMPAT_MSG_RE_C5.match(msg)`?

**Comment by matthewrmshin** [comment]
Looking good. Some minor issues noted. Test OK (with and without site/user global configuration) in my environment apart from the time zone problem in the new test.

@arjclark please review 2.

File: lib/cylc/task_trigger.py (hunk 5 above)
Comment (hjoliver) [inline lib/cylc/task_trigger.py]: ah, yeah, just for `warned`

**Comment by hjoliver** [comment]
All comments addressed.

**Comment by matthewrmshin** [comment]
Tests all good.

**Comment by arjclark** [comment]
Looks ok to me. Test battery passing in my environment.

**Pull request merged**